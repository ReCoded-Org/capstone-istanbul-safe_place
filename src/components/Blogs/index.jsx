import React, { useState, useEffect } from "react";
import { Container, Row } from "react-bootstrap";
import SearchBar from "../SearchBar";
import "./index.scss";

const POSTS_PER_PAGE = 9;
const BLOG_API_URL = `https://public-api.wordpress.com/wp/v2/sites/safeplace102505649.wordpress.com/posts?per_page=${POSTS_PER_PAGE}`;

export default function Blogs() {
  const [blogPosts, setBlogPosts] = useState([]);

  const fetchBlogPosts = async () => {
    const data = await fetch(BLOG_API_URL);
    const fetchedBlogPosts = await data.json();
    setBlogPosts(fetchedBlogPosts);
  };

  useEffect(() => {
    fetchBlogPosts();
  }, []);

  // Search function
  const handleSearch = (e, keyword) => {
    e.preventDefault();
    const filterPost = blogPosts.filter((blog) => {
      const blogTitle = blog.title.rendered;
      const results = blogTitle.toLowerCase().includes(keyword.toLowerCase());
      return results;
    });
    setBlogPosts(filterPost);
  };

  return (
    <Container>
      <SearchBar handleSearch={handleSearch} />
      {blogPosts.map((blogPost) => {
        return (
          <div className="blogsSection" key={blogPost.id}>
            <div className="shadowEffect"></div>
            <h3
              className="blogPostTitle"
              dangerouslySetInnerHTML={{
                __html: blogPost.title.rendered,
              }}
            ></h3>
            <img
              className="blogPostImg"
              alt="Blog post cover"
              src={blogPost.jetpack_featured_media_url}
            />
          </div>
        );
      })}
    </Container>
  );
}
