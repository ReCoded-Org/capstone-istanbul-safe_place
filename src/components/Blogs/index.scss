@import "../../style/variables";

// Bootstrap class
.blogsPage {
  font-family: $mainFontFamily;

  .searchBar {
    margin-bottom: 2rem;
  }
}
