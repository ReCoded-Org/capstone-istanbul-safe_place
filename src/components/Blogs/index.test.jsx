import React from "react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { render, screen, waitFor } from "@testing-library/react";
import Blogs from "./index";
import { act } from "react-dom/test-utils";

let container;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe("Blogs", () => {
  it("Renders without crashing", async () => {
    const mockData = [
      {
        title: {
          rendered: "You can find safe place for yourself here",
        },
        jetpack_featured_media_url:
          "https://safeplace102505649.files.wordpress.com/2020/11/womentogether.jpg",
        id: 78,
      },
    ];

    jest.spyOn(global, "fetch").mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(mockData),
      })
    );

    await act(async () => {
      ReactDOM.render(<Blogs />, container);
    });

    expect(container).toMatchSnapshot();
  });
});
