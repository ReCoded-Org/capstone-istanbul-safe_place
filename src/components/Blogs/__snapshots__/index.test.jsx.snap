// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blogs Renders without crashing 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <input
            class="mr-sm-2 searchBarInput"
            placeholder="Search"
          />
          <img
            alt="Search Icon"
            class="searchIcon"
            src="searchIcon.svg"
          />
        </div>
      </div>
    </div>
    <div
      class="blogsSection"
    >
      <div
        class="shadowEffect"
      />
      <h3
        class="blogPostTitle"
      >
        You can find safe place for yourself here
      </h3>
      <img
        alt="Blog post cover"
        class="blogPostImg"
        src="https://safeplace102505649.files.wordpress.com/2020/11/womentogether.jpg"
      />
    </div>
  </div>
</div>
`;
